src/Video.tsx

```diff
@@ -688,9 +754,7 @@ const Video = forwardRef<VideoRef, ReactVideoProps>(
           }
           viewType={_viewType}
         />
-        {hasPoster && showPoster ? (
-          <Image style={posterStyle} source={{uri: poster}} />
-        ) : null}
+        {hasPoster && showPoster ? _renderPoster() : null}
```

Comment:
I think ideally we should replace this line by `_renderPoster?.()` you need to move has poster && showPoster inside the useCallback !